### PR TITLE
Move jdbc-hana to supported modules

### PR DESCRIPTION
**Pull request by stefanuhrig**

### CODE REVIEW

File: modules/plugin/jdbc/jdbc-hana/pom.xml [hunk 1]
```diff
@@ -17,7 +17,6 @@
     <groupId>org.geotools.jdbc</groupId>
     <artifactId>gt-jdbc</artifactId>
     <version>22-SNAPSHOT</version>
-    <relativePath>../../../plugin/jdbc/pom.xml</relativePath>
```
Comment (jodygarnett) [inline modules/plugin/jdbc/jdbc-hana/pom.xml]: This looks wrong?

File: docs/user/library/jdbc/index.rst [hunk 2]
```diff
@@ -39,6 +39,7 @@ DataStore plugins:
    
    db2
    h2
+   hana
```
Comment (jodygarnett) [inline docs/user/library/jdbc/index.rst]: Do you need to change the hana documentation page at all?

File: modules/plugin/jdbc/jdbc-hana/pom.xml (hunk 1 above)
Comment (stefanuhrig) [reply modules/plugin/jdbc/jdbc-hana/pom.xml]: It is the same as in the other jdbc-plugins. The only have elements groupId, artifactId and version.

File: docs/user/library/jdbc/index.rst (hunk 2 above)
Comment (stefanuhrig) [reply docs/user/library/jdbc/index.rst]: As far as I can tell no further change is necessary. The documentation page is already there (https://docs.geotools.org/latest/userguide/library/jdbc/hana.html) and all the documentation is still up-to-date. Just this file needs an update to move the plugin from the list of unsupported plugins to the supported ones.

Or am I forgetting something?

File: modules/plugin/jdbc/jdbc-hana/pom.xml (hunk 1 above)
Comment (jodygarnett) [reply modules/plugin/jdbc/jdbc-hana/pom.xml]: Sorry I misread the previous, so what was odd was previously this module declared the "wrong" parent :P

File: docs/user/library/jdbc/index.rst (hunk 2 above)
Comment (jodygarnett) [reply docs/user/library/jdbc/index.rst]: Had a look, I like the utility to load additional EPSG definitions.

The pom contains instructions on locally adding the jdbc driver, yep I think we are good.

**Pull request merged**